src/driver_micarray_alsa.cpp

```diff
@@ -17,27 +17,44 @@
 
 #include <iostream>
 
+#include <string>
+
 #include "./driver_micarray_alsa.h"
 
 #include "./src/driver.pb.h"
+
 #include "matrix_hal/microphone_array.h"
 
 namespace matrix_malos {
 
-bool MicArrayAlsaDriver::ProcessConfig(const DriverConfig& /*config*/) {
-  // TODO (andres.calderon@admobilize.com):  handle SetGain
+bool MicArrayAlsaDriver::ProcessConfig(const DriverConfig& config) {
+  MicArrayParams micarray_config(config.micarray());
+
+  const int16_t gain = static_cast<int16_t>(micarray_config.gain());
+
+  mics_->SetGain(gain);
+
+  const float azimutal_angle = micarray_config.azimutal_angle();
+  const float polar_angle = micarray_config.polar_angle();
+  const float radial_distance_mm = micarray_config.radial_distance_mm();
```

Comment:
Do you need the variables? :-) Now that the casts are gone you could just call the following method with the field values directly. Up to you, not asking for a change.